### PR TITLE
Backend APIs For ReadOnly User Permissions

### PR DESCRIPTION
**Pull request by pbardea**
- Currently just a flag for readonly and default to have full permissions.

This is enough for now, i'm not worried about future extensions, but it can easily be changed to an enum.

I can go through again after the demo and clean this up to make it an enum if we want to talk about that aspect of extensibility in our report.

### CODE REVIEW

File: backend/models/users.js [hunk 1]
```diff
@@ -3,10 +3,15 @@ var models = require('../models');
 var uuidv4 = require('uuid/v4');
 var storage = require('../utils/storage');
 
-module.exports.addUserToTripByUserId = (tripId, userId, callback) => {
+module.exports.addUserToTripByUserId = (tripId, userId, readOnly, callback) => {
+  var readOnlyPermission = false;
```
Comment (aneedalie) [inline backend/models/users.js]: You can make this `= readOnly ? true : false`

File: backend/models/users.js [hunk 2]
```diff
@@ -3,10 +3,15 @@ var models = require('../models');
...
+  if (readOnly) {
+  	readOnlyPermission = true;
+  }
   const newRelation = {
-    userId,
-    tripId,
+    userId: userId,
+    tripId: tripId,
+    readOnly: readOnlyPermission,
```
Comment (aneedalie) [inline backend/models/users.js]: Actually you can just put that here instead.

File: backend/routes/users.js [hunk 3]
```diff
@@ -21,26 +21,32 @@ router.get('/:tripId', (req, res) => {
...
+    res.status(200).send(`User ${req.body.userId} added to ${req.body.tripId}.`);
   });
 });
 
 // Add a user email to a given trip
 router.put('/:tripId/email/:userEmail', (req, res) => {
+  var readOnlyPermission = false;
+  if (req.body.readOnly) {
+    readOnlyPermission = true;
+  }
```
Comment (aneedalie) [inline backend/routes/users.js]: Same here as the comments above.

File: backend/models/users.js (hunk 1 above)
Comment (pbardea) [reply backend/models/users.js]: done

**Pull request merged**